tests/integration/targets/test_azure_manage_network_interface/tasks/main.yml

```diff
@@ -85,3 +85,15 @@
         name: "{{ network_interface_virtual_network }}"
         resource_group: "{{ resource_group }}"
         state: absent
+
+    - name: Delete a Network Interface
+      azure.azcollection.azure_rm_networkinterface:
+        name: "{{ network_interface_name }}"
+        resource_group: "{{ resource_group }}"
+        state: absent
+
+    - name: Delete a Network Interface's Autogenerated Security Group
```

Comment:
This one wasn't deleted before?

Reply:
It was added to always section, to clean up in case of failure